src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseCodeReviewFeedbackService.java

```diff
@@ -59,6 +60,10 @@ public class ProgrammingExerciseCodeReviewFeedbackService {
 
     private final ProgrammingMessagingService programmingMessagingService;
 
+    // default value is 3 if the placeholder cannot be resolved
+    @Value("${artemis.athena.allowed-self-learning-feedback-attempts:3}")
```

Comment:
Why is this a property set server-wide by admins? Would it make sense to allow instructors to change this value on a course level? Similar to how it is done for the number of complaints.

Reply:
I think a good solution is to let the admin define the maximum number of attempts in the configuration and also give instructors the possibility to update the number for their specific course so that they can choose between no and the global maximum of allowed requests.

@undernagruzez I think this PR can focus on implementing the global maximum set by the admin in the config. Then, there should be a follow up PR that allows instructors to customize the allowed attempts for their courses.